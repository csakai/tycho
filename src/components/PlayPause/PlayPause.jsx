import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export default class PlayPause extends React.Component {

    static propTypes = {
        onClick: PropTypes.func.isRequired,
        playing: PropTypes.bool
    }

    render() {
        return (
            <div className="play-pause" onClick={this.props.onClick}>
                <span className={cx({
                    'play-pause__button': true,
                    'play-pause__button--playing': this.props.playing,
                    'play-pause__button--paused': !this.props.playing
                })} />
            </div>
        );
    }
}