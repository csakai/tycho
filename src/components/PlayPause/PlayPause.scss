@import 'globals.scss';

.play-pause {
    display: inline-flex;
    cursor: pointer;

    &__button {
        width: $icon-size;
        height: $icon-size;
        background-color: $brand-primary;
        color: $brand-secondary;
        border-radius: 50%;
        margin-right: $base-spacing;
        display: flex;
        align-items: center;
        justify-content: center;

        &--playing {
            &::before {
                content: '\2759\2759';
            }
        }

        &--paused {
            animation: glowing $transition-speed infinite;
            
            &::before {
                content: '\25B6';
            }
        }
    }
}

@keyframes glowing {
  0%   { box-shadow: 0 0 -10px $brand-primary; }
  40%  { box-shadow: 0 0  20px $brand-primary; }
  60%  { box-shadow: 0 0  20px $brand-primary; }
  100% { box-shadow: 0 0 -10px $brand-primary; }
}