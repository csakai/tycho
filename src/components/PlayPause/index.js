import './PlayPause.css';
export {default} from './PlayPause';