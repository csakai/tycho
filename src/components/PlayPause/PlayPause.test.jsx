import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import PlayPause from './';

describe('Play/Pause Button Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<PlayPause onClick={jest.fn()} />);
  });

  it('should render the PlayPause button successfully', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
