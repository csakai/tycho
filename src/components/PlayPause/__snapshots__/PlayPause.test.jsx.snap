// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play/Pause Button Component should render the PlayPause button successfully 1`] = `
<div
  className="play-pause"
  onClick={[Function]}
>
  <span
    className="play-pause__button play-pause__button--paused"
  />
</div>
`;
